src/request/mod.rs

```diff
@@ -43,7 +42,7 @@ pub struct Request<'a, 'b: 'a> {
     pub headers: Headers,
 
     /// The request body as a reader.
-    pub body: Body<'a, 'b>,
+    pub body: Cursor<Vec<u8>>,
```

Comment:
I the hyper client's response interface might be a good thing to look at - I think it's basically a channel of byte buffers?